adr/ADR-001.md

```diff
@@ -0,0 +1,165 @@
+    string max_inflation = 2; // sdk.Dec, percentage: 0 <= x <= 1
+    // min_bonded defines the minimum wanted bond ratio (staked supply/total supply)
+    string min_bonded = 3; // sdk.Dec, percentage: 0 <= x <= 1
+    
+    // max_bonded defines the maximum wanted bond ratio (staked supply/total supply)
+    string max_bonded = 4; // sdk.Dec, percentage: 0 <= x <= 1
+    
+    // inflation_change defines how much the inflation should change if
+    // the bond ratio is not between the defined bands of min/max_bonded.
+    string inflation_change = 5; // sdk.Dec, percentage: 0 <= x <= 1.
+    
+    // max_block_duration defines the maximum duration of a block.
+    // this is important to have because there can be cases for which
+    // a block takes a lot longer than others, for example during upgrades.
+    // In these specific cases we don't want to have bursts of inflation
+    // in a single block.
+    google.protobuf.Duration max_block_duration = 6;
+    
+
+    // InflationRecipient defines the inflation recipients.
+    message InflationRecipient {
+      // recipient is the module receiving inflation, eg: x/staking, x/rewards.
+      string recipient = 1;
+      // ratio is the % of inflation being received. sdk.Dec, percentage: 0 <= x <= 1.
+      string ratio = 2;
+    }
+    
+    // inflation_recipients defines the list of inflation recipients.
+    // CONTRACT: the sum of the ratio of each inflation recipient must be 1.
+    repeated InflationRecipient inflation_recipients = 7;
+}
+```
+
+### State
+
+The state besides `Params` is the following:
+
+Singleton (there can exist only one value, like `Params`). 
+
+```protobuf
+syntax="proto3";
+
+import "google/protobuf/timestamp";
+
+// BlockInfo keeps track of the 
+message LastBlockInfo {
+  // inflation defines the last block's inflation.
+  string inflation = 1; // sdk.Dec, percentage: 0 <= x <= 1
+  // time keeps track of the last block time.  
+  google.protobuf.Timestamp time = 2;
+}
+```
+
+### InitGenesis
+
+
+During InitGenesis, save `Params`, if `LastBlockInfo` is set then use that.
+
+Otherwise `LastBlockInfo` becomes:
+
+```go
+LastBlockInfo{
+	Inflation: Params.MinInflation,
+	Time: ctx.BlockTime(),
+}
+```
+
+### BeginBlock
+
+During begin block:
+
+```go
+package mint
+
+import "time"
+
+const Year = 24 * time.Hour * 365
+
+func BeginBlock(ctx sdk.Context) {
+    lbi := GetLastBlockInfo(ctx)
+    params := GetParams(ctx)
+    timeElapsed := ctx.BlockTime().Sub(lbi.Time)
+    if timeElapsed > params.MaxBlockDuration {
+        timeElapsed = params.MaxBlockDuration
+    }
+
+    totalSupply := GetTotalStakingTokenSupply(ctx)
+    totalBonded := GetTotalStakingTokenSupplyBonded(ctx)
+
+    bondedRatio := totalBonded / totalSupply
+    inflation := lbi.Inflation
+    switch {
+    case bondedRatio < params.MinBondedRatio:
+        inflation += params.InflationChange
+    case bondedRatio > params.MaxBondedRatio:
+        inflation += params.InflationChange
```

Comment:
Should this be that inflation reduces when `bondedRatio < params.MinBondedRatio`

Reply:
this is an incentivisation mechanism for which, if we didn't meet the target bond ratio we start inflating more to convince people to bond more.